datastream-server/src/test/java/com/linkedin/datastream/server/assignment/TestLoadBasedPartitionAssignmentStrategy.java

```diff
@@ -0,0 +1,221 @@
+import com.linkedin.datastream.server.DatastreamGroup;
+import com.linkedin.datastream.server.DatastreamGroupPartitionsMetadata;
+import com.linkedin.datastream.server.DatastreamTask;
+import com.linkedin.datastream.server.DatastreamTaskImpl;
+import com.linkedin.datastream.server.PartitionThroughputInfo;
+import com.linkedin.datastream.server.providers.PartitionThroughputProvider;
+import com.linkedin.datastream.server.zk.KeyBuilder;
+import com.linkedin.datastream.server.zk.ZkAdapter;
+import com.linkedin.datastream.testutil.DatastreamTestUtils;
+import com.linkedin.datastream.testutil.EmbeddedZookeeper;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+
+
+/**
+ * Tests for {@link LoadBasedPartitionAssignmentStrategy}
+ */
+@Test
+public class TestLoadBasedPartitionAssignmentStrategy {
+  private EmbeddedZookeeper _embeddedZookeeper;
+  private ZkClient _zkClient;
+  private String _clusterName;
+
+  @BeforeMethod
+  public void setup() throws IOException {
+    DynamicMetricsManager.createInstance(new MetricRegistry(), "TestStickyPartitionAssignment");
+    _clusterName = "testCluster";
+    _embeddedZookeeper = new EmbeddedZookeeper();
+    String zkConnectionString = _embeddedZookeeper.getConnection();
+    _embeddedZookeeper.startup();
+    _zkClient = new ZkClient(zkConnectionString);
+  }
+
+  @Test
+  public void fallbackToBaseClassWhenElasticTaskAssignmentDisabledTest() {
+    PartitionThroughputProvider mockProvider = mock(PartitionThroughputProvider.class);
+    boolean enableElasticTaskAssignment = false;
+    Optional<Integer> maxTasks = Optional.of(100);
+    Optional<Integer> imbalanceThreshold = Optional.of(50);
+    Optional<Integer> maxPartitionPerTask = Optional.of(100);
+    Optional<Integer> partitionsPerTask = Optional.of(50);
+    Optional<Integer> partitionFullnessFactorPct = Optional.of(80);
+    Optional<Integer> taskCapacityMBps = Optional.of(5);
+    Optional<Integer> taskCapacityUtilizationPct = Optional.of(90);
+    Optional<Integer> throughputInfoFetchTimeoutMs = Optional.of(1000);
+    Optional<Integer> throughputInfoFetchRetryPeriodMs = Optional.of(200);
+    Optional<ZkClient> zkClient = Optional.empty();
+
+    LoadBasedPartitionAssignmentStrategy strategy = new LoadBasedPartitionAssignmentStrategy(mockProvider,
+        maxTasks, imbalanceThreshold, maxPartitionPerTask, enableElasticTaskAssignment, partitionsPerTask,
+        partitionFullnessFactorPct, taskCapacityMBps, taskCapacityUtilizationPct, throughputInfoFetchTimeoutMs,
+        throughputInfoFetchRetryPeriodMs, zkClient, _clusterName);
+
+    Datastream ds1 = DatastreamTestUtils.createDatastreams(DummyConnector.CONNECTOR_TYPE, "ds1")[0];
+    ds1.getSource().setPartitions(0);
+    ds1.getMetadata().put(DatastreamMetadataConstants.TASK_PREFIX, DatastreamTaskImpl.getTaskPrefix(ds1));
```

Comment:
I don't think elasticTaskCount is enabled for the datastream since minTask is not configured on the datastream. To verify, add check if `isElasticTaskAssignmentEnabled` return true for the datastream group.

Reply:
This is the test case for when elastic task assignment is disabled. I don't need it to be enabled.

Reply:
Sorry. You should add check if `isElasticTaskAssignmentEnabled` returns false as well.

I'm not able to find the test where elasticTaskAssignment is enabled and throughputProvider call counts are verified.

Reply:
- `isElasticTaskAssignmentEnabled()` is tested in `fallbackToBaseClassWhenElasticTaskAssignmentDisabledTest(..)` line 95.

- The assertion that `throughputProvider` is called to fetch the data is tested in `updatesNumTasksAndThrowsExceptionWhenNoSufficientTasksTest` line 183.